app/src/androidTest/java/com/android/bookswap/ui/map/MapScreenTest.kt

```diff
@@ -145,224 +140,247 @@ class MapScreenTest {
-  fun displayAllComponents() {
+  fun composeMapScreen(selectedUser: Int = -1) {
     composeTestRule.setContent {
       CompositionLocalProvider(LocalAppConfig provides AppConfig(userViewModel = userVM)) {
         val navController = rememberNavController()
         val navigationActions = NavigationActions(navController)
-        MapScreen(mockBookManagerViewModel, navigationActions, 0)
+        MapScreen(mockBookManagerViewModel, navigationActions, selectedUser, geolocation)
+      }
+    }
+  }
+
+  @Test
+  fun displayAllComponents() {
+    run {
+      step("Compose screen") { composeMapScreen(0) }
+      step("Accept Permissions") {
+        flakySafely { device.permissions.clickOn(Permissions.Button.ALLOW_FOREGROUND) }
+      }
+      step("") {
+        composeTestRule.onNodeWithTag(C.Tag.map_screen_container).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.google_map).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.Marker.info_window_container).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.Marker.info_window_scrollable).assertIsDisplayed()
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.Map.Marker.info_window_book_container)
+            .assertCountEquals(2)
+        composeTestRule.onAllNodesWithTag(C.Tag.Map.Marker.book_title).assertCountEquals(2)
+        composeTestRule.onAllNodesWithTag(C.Tag.Map.Marker.book_author).assertCountEquals(2)
+        composeTestRule.onAllNodesWithTag(C.Tag.Map.Marker.info_window_divider).assertCountEquals(1)
+
+        // components of Draggable Menu
+        composeTestRule.onNodeWithTag(C.Tag.Map.bottom_drawer_container).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.bottom_drawer_layout).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.bottom_drawer_handle).assertIsDisplayed()
+        composeTestRule.onNodeWithTag(C.Tag.Map.bottom_drawer_handle_divider).assertIsDisplayed()
+        composeTestRule
+            .onNodeWithTag("0_" + C.Tag.BookDisplayComp.book_display_container)
+            .assertIsDisplayed()
+        composeTestRule
+            .onNodeWithTag("1_" + C.Tag.BookDisplayComp.book_display_container)
+            .assertIsDisplayed()
+
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.BookDisplayComp.image, useUnmergedTree = true)
+            .assertCountEquals(2)
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.BookDisplayComp.title, useUnmergedTree = true)
+            .assertCountEquals(2)
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.BookDisplayComp.author, useUnmergedTree = true)
+            .assertCountEquals(2)
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.BookDisplayComp.rating, useUnmergedTree = true)
+            .assertCountEquals(2)
+        composeTestRule
+            .onAllNodesWithTag(C.Tag.BookDisplayComp.filled_star, useUnmergedTree = true)
+            .assertCountEquals(9)
```

Comment:
These numbers could perhaps use the variables set above, so the sum of the ratings for the filled stars, the size of the list of books for all above the rating, etc.. This would ensure that if someone changes the dataBook list due to adding more tests, it still works. But I don't think this is important enough.

Reply:
I am also not certain that it is of your own doing, as it is marked as green but there are massive deletions and additions, so it may just be github playing and it was not you to begin with.